src/thor/route_action.cc

```diff
@@ -127,6 +127,27 @@ void remove_edges(const GraphId& edge_id, valhalla::Location& loc, GraphReader&
 namespace valhalla {
 namespace thor {
 
+std::string thor_worker_t::expansion(Api& request) {
+  // tell the path algorithms to track the expansion here
+  expansion_.reset(new rapidjson::Document);
+  // every algorithm gets a default one that is empty
+  expansion_->SetObject();
+  rapidjson::Pointer("/type").Set(*expansion_, "FeatureCollection");
+  rapidjson::Pointer("/properties/algorithm").Set(*expansion_, "none");
+  rapidjson::Pointer("/features/0/type").Set(*expansion_, "Feature");
+  rapidjson::Pointer("/features/0/geometry/type").Set(*expansion_, "MultiLineString");
+  rapidjson::Pointer("/features/0/geometry/coordinates").Create(*expansion_).SetArray();
+  rapidjson::Pointer("/features/0/properties/edge_ids").Create(*expansion_).SetArray();
+  rapidjson::Pointer("/features/0/properties/statuses").Create(*expansion_).SetArray();
+  // fill it out
+  route(request);
+  // serialize it
+  auto json = rapidjson::to_string(*expansion_, 5);
+  // tell the path algorithms to stop tracking the expansion at all
+  expansion_.reset();
```

Comment:
so when we do this reset the algorithms will see that their weak_ptr doesnt point to anything and so wont track